regression/cbmc/gcc_builtin_mul_overflow/main.c

```diff
@@ -0,0 +1,116 @@
+#include <assert.h>
+#include <limits.h>
+#include <stdint.h>
+
+#define A_VALUE_X_SO_THAT_X_TIMES_X_DOES_NOT_OVERFLOW(T)                       \
```

Comment:
That's quite a mouthful :-) Is `MAX_NON_OVERFLOWING_SQRT(T)` any better? Not sure, not a big deal, just floating a suggestion.

Reply:
I went through several different iterations of this name. I didn’t particularly like any one of them, so I settled for the most “honest” one.